arbnode/batch_poster.go

```diff
@@ -467,6 +472,38 @@ func (b *BatchPoster) checkReverts(ctx context.Context, to int64) (bool, error)
 	return false, nil
 }
 
+func (b *BatchPoster) pollForL1PriceData(ctx context.Context) {
+	headerCh, unsubscribe := b.l1Reader.Subscribe(false)
+	defer unsubscribe()
+
+	for {
+		select {
+		case h, ok := <-headerCh:
+			if !ok {
+				log.Info("L1 headers channel checking for l1 price data has been closed")
+				return
+			}
+			baseFeeGauge.Update(h.BaseFee.Int64())
+			if h.BlobGasUsed != nil {
+				if h.ExcessBlobGas != nil {
+					blobFee := eip4844.CalcBlobFee(eip4844.CalcExcessBlobGas(*h.ExcessBlobGas, *h.BlobGasUsed))
+					blobFeeGauge.Update(blobFee.Int64())
```

Comment:
should blobFee be a histogram instead?  I don't know how often value changes...

Reply:
will make it a histogram

Reply:
I think this should be a gauge actually. It changes every 12 seconds and it'd be a lot more useful to have a graph of the exact value over time on a dashboard

Reply:
ah, didn't realize how slow it changed, definitely gauge then